### ISSUE TITLE
CLI fails for a bunch of rules which node API supports

### ISSUE THREAD

**Issue by XOP** [stylelint/stylelint#884]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Here's a bug apparently,
when trying to use linting over CLI it fails.
The same config runs perfectly via node API.

> What CSS is needed to reproduce this issue?

I'm using SCSS syntax

> What stylelint configuration is needed to reproduce this issue?

```json
{
    "rules": {
        "font-family-name-quotes": "double",
        "font-weight-notation": "named",
        "media-feature-no-missing-punctuation": true,
        "no-duplicate-selectors": true,
        "no-unknown-animations": true,
        "time-no-imperceptible":
    }
}
```

> Which version of stylelint are you using?

`4.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

via CLI: (`.stylelintrc` file is in the root folder)
```sh
$ stylelint --syntax scss examples/module.scss
```

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

SCSS syntax (haven't tried with CSS)

> What did you expect to happen?

linting process should not error

> What actually happened (e.g. what warnings or errors you are getting)?

```sh
Error: Undefined rule "font-family-name-quotes"
    at exports.default (/usr/local/lib/node_modules/stylelint/dist/utils/configurationError.js:8:13)
    at /usr/local/lib/node_modules/stylelint/dist/postcssPlugin.js:87:47
    at Array.forEach (native)
    at /usr/local/lib/node_modules/stylelint/dist/postcssPlugin.js:85:33
```
etc. for all other rules

**Comment by XOP** [comment]
hey @jeddy3 , as promised, here's my investigation on bug in CLI

**Comment by jeddy3** [comment]
Thanks for writing this up :)

Can you double check that your *globally* installed stylelint version is definitely `4.4.0` i.e. `stylelint --version`?

**Comment by XOP** [comment]
yeah, it was `4.1.0` :(
everything synced up since installing `4.4.0`

but I wonder, shouldn't it use local npm package instead of global?

**Comment by davidtheclark** [comment]
When you run stylelint's CLI you're going to get whatever version you have of the CLI installed. Having the globally installed CLI run a locally installed version is an atypical additional layer of complexity. Open to PRs about it!
